### PR TITLE
LEG-12: Implement session management infrastructure

### PR DESCRIPTION
**Pull request by sjawhar**
## Summary

Implements the foundational session management system that all authentication features will depend on.

**Files created:**
- `src/legion/session.py` - Complete session management module
- `tests/test_session.py` - 43 tests covering all functionality

**Implemented components:**
1. **Session & SessionConfig dataclasses** - Immutable session records with OWASP-recommended defaults (30min idle, 8hr absolute timeout, 256-bit entropy)
2. **SessionStorage Protocol** - Protocol-based dependency injection for storage abstraction
3. **Token utilities** - Secure token generation, parsing, hashing with path traversal prevention
4. **FileSessionStorage** - File-based storage with atomic writes and 0o700 permissions
5. **create_session** - Create sessions with cryptographically secure tokens
6. **validate_session** - Validate with dual timeout checking
7. **refresh_session** - Update last_accessed_at without extending absolute timeout
8. **revoke_session** - Immediate session invalidation

**Security features:**
- 256-bit entropy via `secrets.token_urlsafe(32)`
- SHA-256 hashing for storage
- Timing-safe comparison via `secrets.compare_digest`
- Path traversal prevention via regex allowlist
- DoS prevention via token length limit (60 chars)
- File permissions restricted to 0o700

## Test Plan
- [x] All 43 session tests pass
- [x] All 144 project tests pass

Fixes LEG-12

### CODE REVIEW

**Comment by sjawhar** [comment]
## Review Summary

**CRITICAL (P1):** 0 issues
**IMPORTANT (P2):** 9 issues
**MINOR (P3):** 12 suggestions

**Verdict: Approved to merge** - The implementation is solid, follows OWASP security guidelines, and meets all stated requirements. The P2 findings are improvements for production-readiness but don't block merging for the current use case (Legion worker authentication).

---

## Review Agents Used
- security-sentinel
- performance-oracle
- architecture-strategist
- pattern-recognition-specialist
- data-integrity-guardian
- kieran-python-reviewer
- agent-native-reviewer
- code-simplicity-reviewer

---

## Key Strengths

**Security (PASS):**
- 256-bit entropy tokens via `secrets.token_urlsafe(32)`
- SHA-256 hashing (never stores raw tokens)
- Timing-safe comparison via `secrets.compare_digest`
- Path traversal prevention via regex allowlist
- DoS prevention via 60-char token limit
- Directory permissions restricted to 0o700

**Architecture (PASS):**
- Clean Protocol-based dependency injection (`SessionStorage`)
- Frozen dataclasses for immutability
- Module-level async functions (matches Legion patterns)
- O(1) session lookup via composite token design

**Agent-Native (PASS):**
- All operations accessible via async APIs
- Proper DI for testing/mocking
- None returns for invalid states (not exceptions)

**Test Coverage (PASS):**
- 43 tests covering all functionality
- Security edge cases tested (path traversal, DoS)
- Expiration scenarios covered

---

## P2 Findings Summary

1. **Race condition in `refresh_session`** - Read-modify-write without locking could cause lost updates under concurrent access
2. **Missing `fsync` before rename** - Power failure during save could result in corrupted session files
3. **Blocking file ops in async** - `Path.rename()` and `Path.unlink()` block the event loop
4. **New storage instance per operation** - Repeated `mkdir`/`chmod` syscalls when using default storage
5. **Double validation in `refresh_session`** - Calls `validate_session` which loads the file, then immediately saves
6. **Imports inside functions** - Stdlib imports (`time`, `json`, `os`) should be at module level
7. **Missing `__all__`** - Public API not explicitly declared
8. **Silent corruption handling** - Corrupted files return `None` without logging
9. **Revoke without token verification** - `revoke_session` only validates format, not hash

---

## Recommended Actions

**Before production deployment:**
- Cache default storage instance to avoid repeated `mkdir`/`chmod`
- Add `fsync` before rename for durability
- Add logging for corrupted session files

**Future improvements:**
- Consider file locking for concurrent access
- Use `anyio.to_thread.run_sync()` for blocking file ops
- Add `list_sessions()` API for session enumeration

File: src/legion/session.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            "last_accessed_at": session.last_accessed_at,
+            "expires_at": session.expires_at,
+            "idle_timeout_secs": session.idle_timeout_secs,
+            "metadata": session.metadata,
+        }
+
+        async with aiofiles.open(temp_path, "w") as f:
+            await f.write(json.dumps(data))
+
+        temp_path.rename(path)  # Atomic on POSIX
```
Comment (sjawhar) [inline src/legion/session.py]: **[P2]:** Missing fsync - Without `f.flush()` and `os.fsync()` before rename, data may not be durably persisted. On crash, the renamed file could contain partial/corrupt data. Consider adding:
```python
await f.flush()
os.fsync(f.fileno())
```

Comment (sjawhar) [inline src/legion/session.py]: **[P2]:** Blocking call - `Path.rename()` is synchronous and blocks the event loop. On slow storage (NFS), this could be 10-100ms. Consider `await anyio.to_thread.run_sync(temp_path.rename, path)`
